grand_challenge_forge/forge.py

```diff
@@ -90,88 +112,164 @@ def generate_archive_cases(
 ):
     result = []
     for i in range(0, number_of_cases):
-        item_files = []
+        item_files = {}
         for input_socket in inputs:
-            zpath = Path(f"case_{i}") / input_socket["relative_path"]
+            # Use deep zpath to create the files
+            zpath = (
+                target_zpath
+                / Path(f"case_{i}")
+                / input_socket["relative_path"]
+            )
 
+            # Report back relative to script paths
             generate_socket_value_stub_file(
                 output_zip_file=output_zip_file,
-                target_zpath=target_zpath / zpath,
+                target_zpath=zpath,
                 socket=input_socket,
             )
 
-            item_files.append(str(zpath))
+            item_files[input_socket["slug"]] = zpath
 
         result.append(item_files)
 
     return result
 
 
+def _interface_context(interfaces):
+    # Build context
+    algorithm_input_sockets = [
+        socket for interface in interfaces for socket in interface["inputs"]
+    ]
+    algorithm_output_sockets = [
+        socket for interface in interfaces for socket in interface["outputs"]
+    ]
+
+    algorithm_interface_keys = []
+    for interface in interfaces:
+        algorithm_interface_keys.append(
+            tuple(sorted([socket["slug"] for socket in interface["inputs"]]))
+        )
+
+    interface_names = [f"interface_{idx}" for idx, _ in enumerate(interfaces)]
+
+    return {
+        "algorithm_interface_names": interface_names,
+        "algorithm_interface_keys": algorithm_interface_keys,
+        "algorithm_input_sockets": algorithm_input_sockets,
+        "algorithm_output_sockets": algorithm_output_sockets,
+    }
+
+
 def generate_example_algorithm(*, output_zip_file, target_zpath, context):
     context = deepcopy(context)
 
+    interface_names = []
+    for idx, interface in enumerate(context["phase"]["algorithm_interfaces"]):
+        interface_name = f"interface_{idx}"
+        interface_names.append(interface_name)
+
+        input_zdir = target_zpath / "test" / "input" / interface_name
+        inputs = interface["inputs"]
+
+        # create inputs.json
+        output_zip_file.writestr(
+            str(input_zdir / "inputs.json"),
+            json.dumps(
+                [socket_to_socket_value(socket) for socket in inputs], indent=4
+            ),
+        )
+
+        # Create actual input files
+        for input in inputs:
+            generate_socket_value_stub_file(
+                output_zip_file=output_zip_file,
+                target_zpath=input_zdir / input["relative_path"],
+                socket=input,
+            )
+
+    # Build context
```

Comment:
Could you just call `_interface_context` here?

Reply:
Ah yes! That was my intention but I totally forgot about it!